kv/client.go

```diff
@@ -177,6 +177,9 @@ func openDB(cc *client.Client, name string, opt badger.Options) (*badger.DB, err
 			if err == nil {
 				break
 			}
+			if err != nil {
+				return nil, err
```

Comment:
its trying several eks in a loop... it seems ignoring the err was proposital... maybe we can check the error types here act accordingly?

also weird that it returns an error and a non-nil db 🤔

Reply:
Indeed, it was returning [here](https://github.com/dgraph-io/badger/blob/master/db.go#L328-L330). I think opening the db shouldn't return an error.